### PR TITLE
refactor: changed tracing init order and settings

### PR DESCRIPTION
**Pull request by CptSchnitz**

### CODE REVIEW

**Review by shimoncohen — APPROVED** [review]

File: src/common/tracing.ts [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+    ignoreOutgoingRequestHook: (request): boolean =>
+      IGNORED_OUTGOING_TRACE_ROUTES.some((route) => typeof request.path === 'string' && route.test(request.path)),
+  },
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  '@opentelemetry/instrumentation-fs': {
+    requireParentSpan: true,
+  },
+});
+
+tracing.start();
```
Comment (shimoncohen) [inline src/common/tracing.ts:18]: Why did you move this? Because of the `node --require`?

File: src/common/tracing.ts [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
-  }),
-  new ExpressInstrumentation(),
-]);
+const tracing = new Tracing(undefined, {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  '@opentelemetry/instrumentation-http': {
+    ignoreIncomingRequestHook: (request): boolean =>
+      IGNORED_INCOMING_TRACE_ROUTES.some((route) => request.url !== undefined && route.test(request.url)),
+    ignoreOutgoingRequestHook: (request): boolean =>
+      IGNORED_OUTGOING_TRACE_ROUTES.some((route) => typeof request.path === 'string' && route.test(request.path)),
```
Comment (shimoncohen) [inline src/common/tracing.ts:10]: Why are you checking `typeof` compared to `INCOMING` where you check for `undefined`?

Comment (CptSchnitz) [reply src/common/tracing.ts:10]: second condition also covers `null`. all based on the typescript types

File: src/common/tracing.ts (hunk 1 above)
Comment (CptSchnitz) [reply src/common/tracing.ts:18]: yes

**Review by netanelC — APPROVED** [review]
